Fortran/UnitTests/defined_io/list-read.f90

```diff
@@ -0,0 +1,65 @@
+module m
+    type base
+        real, allocatable :: data(:)
+        complex, allocatable :: cx(:)
+
+        contains
+
+        procedure :: readBaseFmtd
+        generic :: read(formatted) => readBaseFmtd
+    end type
+
+    contains
+
+    !! read in the array size before allocating dtv's components
+    subroutine readBaseFmtd (dtv, unit, iotype, v_list, iostat, iomsg)
+        class(base), intent(inout) :: dtv
+        integer, intent(in) :: unit
+        character(*), intent(in) :: iotype
+        integer, intent(in) :: v_list(:)
+        integer, intent(out) :: iostat
+        character(*), intent(inout) :: iomsg
+
+        integer isize
+
+        read (unit, *, iostat = iostat, iomsg=iomsg) isize
+        
+        if (allocated(dtv%data)) deallocate (dtv%data)
+
+        allocate (dtv%data(isize))
+
+        read (unit, *, iostat=iostat, iomsg=iomsg) dtv%data
+        read (unit, *, iostat = iostat, iomsg=iomsg) isize
+        if (allocated(dtv%cx)) deallocate (dtv%cx)
+
+        allocate (dtv%cx(isize))
+
+        read (unit, *, iostat=iostat, iomsg=iomsg) dtv%cx
+    end subroutine
+end module
+
+program list_read
+    use m
+    integer currentPos, ss
+
+    type(base) :: b1(2)
+
+
+    open (1, file='dcmlChildRead003.data', access='stream', form='formatted', &
```

Comment:
This OPEN statement should use a file name that is test name dependent, and have a STATUS= specifier of 'scratch'.